### ISSUE TITLE
make unit-lib fails locally with "Failed to run protoc"

### ISSUE THREAD

**Issue by baumanj** [habitat-sh/builder#498]
```
➤ env RUST_BACKTRACE=1 make unit-lib
cat: VERSION: No such file or directory
sh -c 'cd components/builder-db && cargo test'
   Compiling habitat-builder-protocol v0.0.0 (file:///home/jbauman/builder/components/builder-protocol)
error: failed to run custom build command for `habitat-builder-protocol v0.0.0 (file:///home/jbauman/builder/components/builder-protocol)`
process didn't exit successfully: `/home/jbauman/builder/target/debug/build/habitat-builder-protocol-d8de4e04d612121c/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'Failed to run protoc, please check that it is available on your PATH, and that the src/message folder is writable: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }', libcore/result.rs:945:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:207
   3: std::panicking::default_hook
             at libstd/panicking.rs:223
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:402
   5: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:349
   6: rust_begin_unwind
             at libstd/panicking.rs:325
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:72
   8: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:26
   9: <core::result::Result<T, E>>::expect
             at /checkout/src/libcore/result.rs:809
  10: build_script_build::protocols::generate_protocols
             at components/builder-protocol/../libbuild-protocols.rs:11
  11: build_script_build::main
             at components/builder-protocol/build.rs:5
  12: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  13: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:306
  14: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
  15: std::rt::lang_start_internal
             at libstd/panicking.rs:285
             at libstd/panic.rs:361
             at libstd/rt.rs:58
  16: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  17: main
  18: __libc_start_main
  19: _start

Makefile:123: recipe for target 'unit-builder-db' failed
make: *** [unit-builder-db] Error 101
```

The `cat: VERSION: No such file or directory` line is unrelated to this failure and will be addressed in [a separate PR](https://github.com/habitat-sh/builder/pull/497)

**Comment by baumanj** [comment]
I believe this is a permission issue since it gets further when run as root, but there are other issues as well. For the sake of usability, the make commands should either work locally or make clear that they are not suitable for local use.

**Comment by chefsalim** [comment]
The issue here is that when protoc is not present, the protocols will not compile.  So the issue is not make unit-lib (which works ok).  Since we moved to the studio for local development, the scripts and instructions that were needed to get a local 'cargo build' functional from scratch were removed. There was a separate related issue to address this (closed now) - https://github.com/habitat-sh/builder/issues/344.  At some point we may bring that back to address the overall issue.
